### ISSUE TITLE
Actor test hanging.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#362]
Fairly frequently, I'll see

```
python test/actor_test.py ActorMethods.testDefineActor
```

hang. An example log from Travis is below. https://s3.amazonaws.com/archive.travis-ci.org/jobs/210142084/log.txt

```
testDefineActor (__main__.ActorMethods) ... 11263:M 11 Mar 22:14:56.612 # Server started, Redis version 3.9.102

Waiting for redis server at 127.0.0.1:65462 to respond...

[INFO] (/Users/travis/build/ray-project/ray/src/plasma/plasma_store.cc:868) Allowing the Plasma store to use up to 3.44GB of memory.

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11276

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11277

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11278

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11279

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11280

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11281

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11289

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11295

Not starting the web UI because the web UI requires Python 3.

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11297

[INFO] (/Users/travis/build/ray-project/ray/src/local_scheduler/local_scheduler.cc:232) Started worker with pid 11304



No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```

**Comment by robertnishihara** [comment]
One possibility is the following:

- We notify a local scheduler that it is supposed to create an actor with a redis command like `PUBLISH actor_notifications <actor_id><local_scheduler_id>`.
- The local scheduler subscribes to this channel and passes in a callback that creates the actor in response to a message arriving on the channel.
- However, if the publish happens before the subscription is in place, then the actor will never be created and an invocation of one of the actor's methods will never execute.

If this is the problem, then it's part of a recurring problem in which we're using Redis's pubsub functionality as a message bus, but when a client subscribes to a channel, it only receives messages that are sent after the subscription. If it wants to receive previous messages, another mechanism or a layer is needed.

**Comment by robertnishihara** [comment]
This happened again here https://s3.amazonaws.com/archive.travis-ci.org/jobs/214857836/log.txt.

**Comment by robertnishihara** [comment]
This may have been resolved by #513.
